pkg/apis/backupdriver/v1/backup_repository.go

```diff
@@ -0,0 +1,47 @@
+package v1
+
+import meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+/*
+ BackupRepository is a cluster-scoped resource.  It is controlled by the Backup Driver and referenced by
+ Snapshot, CloneFromSnapshot and Delete.  The BackupRespository resource contains the credential for a backup repository.
+ The RepositoryDriver defines the driver that will be used to talk to the repository
+ Only Snapshot,etc. CRs from namespaces that are listed in AllowedNamespaces will be acted on, if the namespace is
+ not in AllowedNamespaces the operation will fail.
+*/
+type BackupRepository struct {
+	// TypeMeta is the metadata for the resource, like kind and apiversion
+	meta_v1.TypeMeta `json:",inline"`
+
+	// +optional
+	meta_v1.ObjectMeta `json:"metadata,omitempty"`
+
+	AllowedNamespaces []string `json:"allowedNamespaces"`
+
+	RepositoryDriver      string            `json:"repositoryDriver"`
+	Credential            map[string]string `json:"credential"`
+	BackupRepositoryClaim string            `json:backupRepositoryClaim`
+}
+
+/*
+ BackupRepositoryClaim is used to define/access a BackupRepository.  A new BackupRepository will be created
+ with the RepositoryDriver, Credential and AllowedNamespaces will either be the namespace that the BackupRepositorySpec
+ was created in or the AllowedNamespaces specified in the BackupRepositorySpec.  The BackupRepository field will
```

Comment:
s/BackupRepositorySpec/BackupRepositoryClaim

Reply:
I don't quite understand this.  Is BackupRepositoryClaim a namespaced object?  If it is a namespaced object, then having AllowedNamespaces in BackupRepositoryClaim seems really odd.

Reply:
Well, the alternative is to have Velero write BackupRepositoryClaims into each namespace that it is going to backup and have a BackupRepository per namespace.  Maybe it doesn't matter but it seemed like a bunch more objects to manage.